test/metabase/query_processor_test/timezones_test.clj

```diff
@@ -297,42 +297,56 @@
               (doseq [[expected-row row] (map vector expected-rows rows)]
                 (is (= expected-row row))))))))))
 
-(deftest filter-datetime-by-date-in-timezone-test
+(deftest filter-datetime-by-date-in-timezone-relative-to-current-date-test
   (mt/test-drivers (set-timezone-drivers)
     (testing "Relative to current date"
       (let [expected-datetime (u.date/truncate (t/zoned-date-time) :second)]
-        (mt/with-temp-test-data
-            ["relative_filter"
-             [{:field-name "created", :base-type :type/DateTimeWithTZ}]
-             [[expected-datetime]]]
-            (doseq [timezone ["UTC" "America/Los_Angeles"]]
-              (mt/with-temporary-setting-values [report-timezone timezone]
-                (is (= (-> expected-datetime
-                           (u.date/with-time-zone-same-instant timezone)
-                           t/offset-date-time)
-                       (-> (mt/run-mbql-query relative_filter {:fields [$created]
-                                                               :filter [:time-interval $created :current :minute]})
-                           mt/first-row
-                           first
-                           (u.date/parse nil)
-                           t/offset-date-time))))))))
+        (mt/with-temp-test-data ["relative_filter"
+                                 [{:field-name "created", :base-type :type/DateTimeWithTZ}]
+                                 [[expected-datetime]]]
+          (doseq [timezone ["UTC" "America/Los_Angeles"]]
+            (mt/with-temporary-setting-values [report-timezone timezone]
+              (let [query (mt/mbql-query relative_filter {:fields [$created]
+                                                          :filter [:time-interval $created :current :minute]})]
+                (mt/with-native-query-testing-context query
+                  (let [results (qp/process-query query)]
+                    (is (=? {:status :completed}
+                            results))
+                    (is (= (-> expected-datetime
+                               (u.date/with-time-zone-same-instant timezone)
+                               t/offset-date-time)
+                           (some-> results
+                                   mt/first-row
+                                   first
+                                   (u.date/parse nil)
+                                   t/offset-date-time)))))))))))))
+
+(deftest filter-datetime-by-date-in-timezone-relative-to-days-since-test
```

Comment:
I split this one mega-test into three separate tests. There's no real reason it should all be one test, unless there is a lot of shared setup state or something it's probably better to have finer-grained tests. At least it's easier to debug